errors/errors.go

```diff
@@ -24,16 +28,28 @@ const (
 
 // BoulderError represents internal Boulder errors
 type BoulderError struct {
-	Type   ErrorType
-	Detail string
+	Type      ErrorType
+	Detail    string
+	SubErrors []SubBoulderError
+}
+
+// TODO(@cpu): SubBoulderError type docstring
+type SubBoulderError struct {
+	BoulderError
+	Identifier identifier.ACMEIdentifier
 }
 
-func (be *BoulderError) Error() string {
+func (be BoulderError) Error() string {
```

Comment:
Having a pointer receiver on `Error` was preventing `BoulderError` from implementing `error`.

Reply:
> Right, but *BoulderError did implement error. Why did you need the non-pointer form to implement error instead? 

I don't think this is true. An `Error()` function with a pointer receiver does **not** implement `error`. Here's a Play example: https://play.golang.com/p/yGUznG_8tp1

The pointer receiver needs to be removed.

Reply:
This works fine if you return a `*foo` (i.e `return &foo{}): https://play.golang.com/p/1fQT_QCB-9A

Reply:
Ahhhhh! I see what you mean now. Thanks!